xdsl/dialects/mpi.py

```diff
@@ -468,6 +501,38 @@ def get(request: Operand, ignore_status: bool = True):
         return Wait.build(operands=[request], result_types=result_types)
 
 
+@irdl_op_definition
+class WaitAll(MPIBaseOp):
```

Comment:
nit: if we're being consistent with the MPI names, then maybe this should be called `Waitall`
```suggestion
class Waitall(MPIBaseOp):
```